### PR TITLE
Explicitly copy sample data and info when copying a record

### PR DESCRIPTION
**Pull request by anju24**

### CODE REVIEW

File: scripts/liftover.py [hunk 1]
```diff
@@ -219,7 +219,14 @@ def convert_grch38_ref_mismatch_sites_to_grch37(input_vcf_file, output_vcf_basen
             mismatch_record.QUAL = 100
             mismatch_record.FILTER = []
             mismatch_record.FORMAT = 'GT'
+            mismatch_record.samples = []
+            # copy the objects within a record.
+            # Without doing an explicit copy it will just be a
+            # pointer to the original record
+            for sample in record.samples:
+                mismatch_record.samples.append(copy(sample))
             mismatch_record.samples[0].data = calldata_spec('1/1')
```
Comment (wongkarenhy-color) [inline scripts/liftover.py:228]: Do we need to worry about multi sample vcf? If so we'd need to update `calldata_spec` for all samples.

**Review by wongkarenhy-color — APPROVED** [review]
Lgtm! Just a couple of comments.

File: scripts/liftover.py (hunk 1 above)
Comment (anju24) [reply scripts/liftover.py:228]:
```In [16]: record
Out[16]: <vcf.model._Record at 0x11ad0b780>

In [17]: record_copy = deepcopy(record)

In [18]: record.samples
Out[18]: [Call(sample=sample1, CallData(GT=0/1))]

In [19]: record_copy.samples
Out[19]: [Call(sample=sample1, <class 'vcf.model.make_calldata_tuple.<locals>.CallData'>)]```

At multiple depths deep copy does not seem to be copying the exact instance.

File: scripts/liftover.py [hunk 2]
```diff
@@ -219,7 +219,14 @@ def convert_grch38_ref_mismatch_sites_to_grch37(input_vcf_file, output_vcf_basen
             mismatch_record.QUAL = 100
             mismatch_record.FILTER = []
             mismatch_record.FORMAT = 'GT'
+            mismatch_record.samples = []
+            # copy the objects within a record.
+            # Without doing an explicit copy it will just be a
+            # pointer to the original record
+            for sample in record.samples:
+                mismatch_record.samples.append(copy(sample))
```
[the remark this answers is not in the document]
Comment (anju24) [reply scripts/liftover.py:227]: Out[16]: <vcf.model._Record at 0x11ad0b780>

In [17]: record_copy = deepcopy(record)

In [18]: record.samples
Out[18]: [Call(sample=sample1, CallData(GT=0/1))]

In [19]: record_copy.samples
Out[19]: [Call(sample=sample1, <class 'vcf.model.make_calldata_tuple.<locals>.CallData'>)]```

At multiple depths deep copy does not seem to be copying the exact instance.

File: scripts/liftover.py (hunk 1 above)
Comment (anju24) [reply scripts/liftover.py:228]: No. I think we are ok. There will only be one sample per vcf in these files.

Comment (anju24) [reply scripts/liftover.py:228]: But I will add that in the README to be sure.

**Pull request merged**